source/AnimatedEntity.cpp

```diff
@@ -74,3 +76,34 @@ void AnimatedEntity::updateAnimations(Ogre::Real r)
     }
   }
 }
+
+void AnimatedEntity::setMount(AnimatedEntity *m)
+{
+  if (entityMount)
+    dismount();
+  entityMount.reset(m);
+  if (entityMount)
+    dismount();
+}
+
+void AnimatedEntity::mount(void)
+{
+  if (entityMount && isMounted() == false)
```

Comment:
is the checking `AnimatedEntity::isMounted()` necessary?
Also generally `== false` is bad style.